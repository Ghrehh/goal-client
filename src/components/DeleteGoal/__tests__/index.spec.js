import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow'
import { Delete } from '../';
import { mount } from 'enzyme';

describe('Delete', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};
    testContext.auth = { userId: 1, rememberToken: 'rememberToken' }
    testContext.deleteGoalMock = jest.fn()

    testContext.component = (
      <Delete
        auth={testContext.auth}
        deleteGoal={testContext.deleteGoalMock}
        goalId={1}
        loading={false}
        error={undefined}
        called={false}
      />
    );
  });

  describe('snapshot', () => {
    beforeEach(() => {
      const renderer = new ShallowRenderer();
      testContext.result = renderer.render(testContext.component);
    });

    it('renders the component correctly', () => {
      expect(testContext.result).toMatchSnapshot();
    });
  });

  describe('deleting', () => {
    beforeEach(() => {
      const wrapper = mount(testContext.component);
      wrapper.simulate('click');
    });

    it('calls the delete goal mock with the correct arguments', () => {
      expect(testContext.deleteGoalMock).toBeCalledWith({
        variables: { goalId: 1, auth: testContext.auth }
      });
    });
  });
});
