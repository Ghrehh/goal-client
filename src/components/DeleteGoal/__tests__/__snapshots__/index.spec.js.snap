// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete snapshot renders the component correctly 1`] = `
<LoadingAndErrorHandler
  loading={false}
>
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</LoadingAndErrorHandler>
`;
